spyder/widgets/mixins.py

```diff
@@ -1359,9 +1363,17 @@ def set_help_enabled(self, state):
         self.help_enabled = state
 
     def inspect_current_object(self):
-        text = self.get_current_word(help_req=True)
-        if text:
-            self.show_object_info(text, force=True)
+        regex_word = ''
+        start_text = self.get_text('sol', 'cursor')
+        tl1 = re.findall(r'([a-zA-Z_]+[^\s^\(]*)', start_text)
+        if tl1 and start_text.endswith(tl1[-1]):
+            regex_word += tl1[-1]
+        end_text = self.get_text('cursor', 'eol')
+        tl2 = re.findall(r'([^\s^\(]+[0-9a-zA-Z_\.]*)', end_text)

```

Comment:
How many words with dots this regexp can capture? Please remember that we can have things like `np.linalg.norm` too.

Also, perhaps this should only apply to Python and not to other languages (e.g. the equivalent of `.` in R is `$`).

Reply:
Also, why can't we use here the current text from the cursor to the left, until an space is found? That seems simpler to me than using regexp's.

Reply:
I tried that approach at first, but then I realized that I needed also to get sometimes characters to the right of the cursor, and I wasn't sure of the logic to follow to get does characters and then mix both (left and right selected texts).

I will recheck that approach and see what is better

Reply:
Thanks @dalthviz!